ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -352,6 +355,9 @@ runWith RunNodeArgs{..} LowLevelRunNodeArgs{..} =
                                     nodeKernel
                                     peerMetrics
 
+      atomically $ modifyTVar shouldCreateCleanMarker (const True)
+      cleanRemover
```

Comment:
What is the `cleanRemover` for, and how does it interact with `shouldCreateCleanMarker`?

Reply:
Why was this not needed before? I mean, why didn't we need to create the clean marker after successfully opening the chain DB?

Reply:
The TVar means whether it should create a new clean file in presence of a non-corrupting exception. The function is the one to remove the old clean file.

It is still created on exit by `withCheckedDb`. We are now delaying the deletion of the previous one until we opened the DB. Notice that I captured the IO action for deleting it into the `cleanRemover`

Reply:
In both the old code and the new, the marker is created by the `bracket`-like closing part of `llrnWithCheckedDB`. In Javier's PR here, the new bit is the reference which might prevent the marker from being created on a clean shutdown.